src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -27,13 +27,16 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.concurrent.locks.ReentrantLock;
 
 import static io.nats.client.support.BuilderBase.bufferAllocSize;
-import static io.nats.client.support.NatsConstants.CR;
-import static io.nats.client.support.NatsConstants.LF;
+import static io.nats.client.support.NatsConstants.*;
 
 class NatsConnectionWriter implements Runnable {
+    enum Mode {
+        Normal, Reconnect, WaitingForEndReconnect
+    }

```

Comment:
Normal - regularly connected
Reconnect - Used during reconnect for internal messages like connect, ping, and re-subscribe 
WaitingForEndReconnect - So we still pull from the reconnectQueue, but all messages published after this go into the normal queue